javaslang/src/main/java/javaslang/collection/JavaConverters.java

```diff
@@ -0,0 +1,451 @@
+            return ListIterator.of(getDelegate(), index, isMutable());
+        }
+
+        @Override
+        public T remove(int index) {
+            // may throw an IndexOutOfBoundsException accordingly to the j.u.List.remove(int)
+            return setDelegateAndGetPreviousElement(index, getDelegate().removeAt(index));
+        }
+
+        @Override
+        public boolean remove(Object obj) {
+            // may throw a ClassCastException accordingly to the j.u.List.remove(Object)
+            @SuppressWarnings("unchecked") final T that = (T) obj;
+            return setDelegateAndCheckChanged(getDelegate().remove(that));
+        }
+
+        @Override
+        public boolean removeAll(Collection<?> collection) {
+            // may throw a ClassCastException accordingly to the j.u.List.removeAll(Collection)
+            @SuppressWarnings("unchecked") final Collection<T> that = (Collection<T>) collection;
+            return setDelegateAndCheckChanged(getDelegate().removeAll(that));
+        }
+
+        @Override
+        public boolean retainAll(Collection<?> collection) {
+            // may throw a ClassCastException accordingly to j.u.List.retainAll(Collection)
+            @SuppressWarnings("unchecked") final Collection<T> that = (Collection<T>) collection;
+            return setDelegateAndCheckChanged(getDelegate().retainAll(that));
+        }
+
+        @Override
+        public T set(int index, T element) {
+            // may throw an IndexOutOfBoundsException accordingly to the j.u.List.set(int, T)
+            return setDelegateAndGetPreviousElement(index, getDelegate().update(index, element));
+        }
+
+        @Override
+        public int size() {
+            return getDelegate().size();
+        }
+
+        @Override
+        public List<T> subList(int fromIndex, int toIndex) {
+            // may throw IndexOutOfBoundsException accordingly to j.u.List.subList(int, int)
+            return new ListView<>(getDelegate().subSequence(fromIndex, toIndex), isMutable());
+        }
+
+        @Override
+        public Object[] toArray() {
+            return getDelegate().toJavaArray();
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public <U> U[] toArray(U[] array) {
+            final U[] target;
+            final int length = getDelegate().length();
+            if (array.length < length) {
+                final Class<? extends Object[]> newType = array.getClass();
+                target = (newType == Object[].class)
+                         ? (U[]) new Object[length]
+                         : (U[]) java.lang.reflect.Array.newInstance(newType.getComponentType(), length);
+            } else {
+                target = array;
+            }
+            final Iterator<T> iter = iterator();
+            for (int i = 0; i < length; i++) {
+                target[i] = (U) iter.next();
+            }
+            return target;
```

Comment:
Don't forget to set a `null` element if the array is oversized: https://docs.oracle.com/javase/8/docs/api/java/util/List.html#toArray-T:A-